xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -900,7 +900,10 @@ bool CGUIMediaWindow::OnClick(int iItem, const std::string &player)
     else if (!CProfilesManager::GetInstance().GetCurrentProfile().canWriteSources() && !g_passwordManager.IsProfileLockUnlocked())
       return false;
 
-    return OnAddMediaSource();
+    if (OnAddMediaSource())
+      Refresh();
+
+    return true;
```

Comment:
this is something completely diffent than the filecache arnova is working on. a cache does not need to be cleared after an item was added. if this was the case, the cache design would be completely wrong.